### ISSUE TITLE
Relax Tie._attachment_test_all() restrictions.

### ISSUE THREAD

**Issue by josiah-wolf-oberholtzer** [Abjad/abjad#758]
Tie attachment tests should allow for tied chords which are sub/supersets of each other. That is (apparently) valid LilyPond code.

From Fredrik Wallberg:

---

Hi List, I just noticed that Abjad 2.18 does not allow ties across chords containing different pitches.

```
from abjad import *
staff = Staff("<c' e' g'>4 <c' e'>4")
attach(Tie(), staff[:])
```

...results in:
 
```
Exception: Tie() attachment test fails for Selection(Chord("<c' e' g'>4"), Chord("<c' e'>4"))
```

I'm probably overlooking something obvious, but how would you model a chord sequence which results in the following LilyPond?

```
\new Staff {
    <c' e' g'>4 ~
    <c' e'>4
}
```

Thanks! Best / Fredrik

**Comment by quesebifurcan** [comment]
Thanks for looking into this. Here's another example which shows why a simple pitch equality check doesn't quite cut it:

```
\new Staff {
  <c'> ~ <c' d'> ~ <d'> ~ <d' e'>
}
```

Just FYI, this is what I did to make it work (in `abjad/tools/spannertools/Tie.py`)

```python
def _attachment_test_all(self, component_expression):
    from abjad.tools import scoretools
    from abjad.tools import sequencetools
    written_pitches = []
    if isinstance(component_expression, scoretools.Component):
        component_expression = [component_expression]
    for component in component_expression:
        if isinstance(component, scoretools.Note):
            written_pitches.append(set([component.written_pitch]))
        elif isinstance(component, scoretools.Chord):
            written_pitches.append(set(component.written_pitches))
        else:
            return False
    for pair in sequencetools.iterate_sequence_nwise(written_pitches):
        if not set.intersection(*pair):
            return False
    return True
```

(Not familiar with the implications of such a change, so take it with a grain of salt)

**Comment by josiah-wolf-oberholtzer** [comment]
@quesebifurcan That solution looks like what I would do too. Since it works, just send over a PR. I'll merge it ASAP.

**Comment by josiah-wolf-oberholtzer** [comment]
Fixed in #762.
